index.bs

```diff
@@ -118,3 +118,39 @@ The attacker writes a malicious `text/html` file to the server. Depending on the
 #### Considerations
 
 Servers are strongly encouraged to consider the countermeasures in the context of the use cases they want to enable or disable on a given storage. For instance, using `Content-Security-Policy: sandbox` will universally prohibit various functionalities for applications, including but not limited to accessing local storage, executing scripts, using forms, interacting with plugins, or including external content. This broad range of restrictions may not be desirable for various categories of applications that rely on client-side storage mechanisms, collaborative features, or dynamic content interaction.
+
+## Protecting WebID Document ## {#protecting-webid-document}
+
+Solid access policy enforcement heavily relies on the identities of agents. Requesting Parties are identified
+with [[WebID]] and applications with ClientID [[Solid.OIDC]].
+User's WebID Document includes trust anchors, like designation to their [[Solid.OIDC]] Provider (aka. issuer).
+In other approaches, public keys could be published or discoverable via the user's WebID Document.
+
+### Fully impersonate the user
+
+If the malicious app can write or append to the user's WebID, it could inject trust anchors, allowing complete impersonation.
+
+#### Prerequisites
+
+* Authorization system depends on a specific trust anchor in the user's WebID Document.
+    For example, [[Solid.OIDC]] issuer designation or pubic keys/keyset.
+* User authorizes a malicious application to write or append to their WebID Document.
+
+#### Attack
+
+* The attacker publishes a malicious application, such as generating cool AI avatars and setting them in the user's profile.
+* User authenticates with the malicious application
+  * If the access control system enforces client constraints, the user also authorizes the application to write or append to their WebID Document.
+* Application injects attacker's trust anchor into the user's WebID Document 
+* The attacker can completely impersonate the user, which allows full access to all the data owned by the user.
+    As well as any data shared with that user by others, with the full extent of access they granted to the user.
+
+### Countermeasures
+
+* There is no requirement to expose WebID Document via [[Solid.Protocol]] and host it in Solid Storage.
```

Comment:
This needs more nuance. Technically true for *any* WebID Profile Document but not true for Solid WebID Profiles. In other words, WebID Profile Documents served from a Solid Storage - which is a core requirement from Solid use cases - naturally use the Solid Protocol.

Reply:
Solid Protocol 0.11 doesn't depend on a WebID profile draft. If Solid Protocol 0.11 requires a WebID Document to be exposed via the Solid Protocol, please link it in this thread.

Reply:
```suggestion
* There is no requirement to expose WebID Document via [[Solid.Protocol]] nor to host it in Solid Storage.
```

Reply:
@elf-pavlik , in https://github.com/solid/security-considerations/pull/9/files#r1626487426 you are now using *Solid application* as the prerequisite of the attack. That's obviously only meaningful in the context of the Solid Protocol and hence Solid storage since Solid applications can't be guaranteed to do anything against servers that do not conform to the Solid Protocol.

If the idea is to only detail attacks pertaining to WebID Profile Documents hosted on a Solid storage and that their trust anchors could change, then stick to that.

Reply:
Hosting a WebID Document in Solid Storage makes this attack possible. One proposed countermeasure is not to host the WebID Document in Solid Storage. 

* the attack relies on the WebID Document being hosted in a Solid Storage
* one of the countermeasures relies on WebID Document **not** being hosted in a Solid Storage

Reply:
So, the take away is that if we don't use Solid, the system will actually be more secure? =) Maybe we need to rethink that...

A similar attack categorically exists when the document is not hosted on a Solid storage. Whatever takes to modify the trust anchor.

Reply:
Please consider comparing it to a secured house with a security safe inside. While the house already has a security system, some critical items are protected by being placed in a security safe. Does it make no sense to use the safe if you can't fit the whole house in it?
We agreed at the last meeting that what is being discussed here is an actual security issue existing in many Solid deployments. What's proposed here is only one possible solution. Please feel free to PR another solution to the vulnerability we talked about here. I'm currently implementing the proposed solution as an option for CSS; It is not required, but I hope you or someone else plans to work on implementing the alternative, which I assume you would like to add in a dedicated PR.

Reply:
> So, the take away is that if we don't use Solid, the system will actually be more secure? =) Maybe we need to rethink that...

Right. I think this PR is saying "if you need to host a WebID Document securely, don't do it with Solid". What I think we *should* be doing here is "If you need to host a WebID Document securely *on a Solid storage*, here's how to do it".

Reply:
Those two approaches are not mutually exclusive. One may be available today, and one could be made available sometime in the future.

Reply:
> So, the take away is that if we don't use Solid, the system will actually be more secure? =) Maybe we need to rethink that...
> 

We also need to rethink removing user control over the most important document in the Solid ecosystem.  @elf-pavlik mentioned how many members of the CG have  their WebIDs hosted on non-Solid servers but that is really irrelevant.  A CG member does not lose control over their WebID, they know how to edit it and what the edits mean.  This does not apply to the general public who could benefit from a profile-editing app that informs them of what their changes mean.